src/lexer.c

```diff
@@ -403,14 +433,83 @@ void tryParseExportStatement () {
 
 char16_t readExportAs (char16_t* startPos, char16_t* endPos) {
   char16_t ch = *pos;
+
+  bool startWithQuote = false;
+  bool withAs = false;
+  // export { "identifer" as } from
+  // export { "@notid" as } from
+  // export { "spa ce" as } from
+  // export { " space" as } from
+  // export { "space " as } from
+  // export { "not~id" as } from
+  // export { "%notid" as } from
+  // export { "identifer" } from
+  // export { "%notid" } from
+  if (isQuote(ch)) {
+    startWithQuote = true;
+    char16_t end_quote;
+    while ((end_quote = *(++pos)) != '}') {
+      if (end_quote == ch && *(pos - 1) != '\\') {
+        break;
+      }
+    }
+
+    if (end_quote == '}') {
+      syntaxError();
+      return ch;
+    }
+
+    readToWsOrPunctuator(ch);
+    endPos = pos;
+    withAs = str_eq2(pos + 1, 'a', 's');
+  }
```

Comment:
This quote scanning should move up into its own function / into the parent, then we can just keep the as logic below.